### PR TITLE
Remove ingest processor supports excluding fields

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
The remove ingest processor will support excluding fields in version 2.12.0, a new parameter `exclude_filed` is added to that processor, this PR updates the remove processor's document to add the parameter.

### Issues Resolved
Fixes https://github.com/opensearch-project/documentation-website/issues/5918

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ingest-pipelines/processors/remove.md [hunk 1]
```diff
@@ -30,7 +30,8 @@ The following table lists the required and optional parameters for the `remove`
 
 | Parameter  | Required/Optional  | Description  |
 |---|---|---|
-`field`  | Required  | The name of the field containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields are not allowed to be removed: `_index`, `_version`, `_version_type`, and `_id`. Note that `_id` is not allowed to be removed when there's a specified external version for the ingesting document. |
+`field`  | Optional  | The name of the field containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields are not allowed to be removed: `_index`, `_version`, `_version_type`, and `_id`. Note that `_id` is not allowed to be removed when there's a specified external version for the ingesting document. |
```
Comment (vagimeli) [inline _ingest-pipelines/processors/remove.md:33]:
```suggestion
`field`  | Optional  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields cannot be removed: `_index`, `_version`, `_version_type`, and `_id`. If a version is specified for the ingesting document, the `_id` cannot be removed. |
```

File: _ingest-pipelines/processors/remove.md [hunk 2]
```diff
@@ -30,7 +30,8 @@ The following table lists the required and optional parameters for the `remove`
...
+`exclude_field`  | Optional  | The name of the fields need to be kept while other fields except the metadata fields will be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). `exclude_field` and `field` are mutually exclusive, either of them can be set. |
```
Comment (vagimeli) [inline _ingest-pipelines/processors/remove.md:34]:
```suggestion
`exclude_field`  | Optional  | The field name to retained. All other fields, except metadata fields, will be removed. The `exclude_field` and `field` options are mutually exclusive. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).  |
```

**Review by vagimeli — APPROVED** [review]
@gaobinlong Please see the doc review edits. I'll move this into editorial once changes are made. Thank you!

File: _ingest-pipelines/processors/remove.md [hunk 3]
```diff
@@ -30,7 +30,8 @@ The following table lists the required and optional parameters for the `remove`
 
 | Parameter  | Required/Optional  | Description  |
 |---|---|---|
-`field`  | Required  | The name of the field containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields are not allowed to be removed: `_index`, `_version`, `_version_type`, and `_id`. Note that `_id` is not allowed to be removed when there's a specified external version for the ingesting document. |
+`field`  | Optional  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). The following metadata fields cannot be removed: `_index`, `_version`, `_version_type`, and `_id`. If a version is specified for the ingesting document, the `_id` cannot be removed. |
```
Comment (natebower) [inline _ingest-pipelines/processors/remove.md:33]: As written, "ingesting document" implies that the document is ingesting something. Is this accurate, or should we reword?

**Review by natebower — COMMENTED** [review]
@vagimeli Just one comment. Thanks!

File: _ingest-pipelines/processors/remove.md (hunk 3 above)
Comment (vagimeli) [reply _ingest-pipelines/processors/remove.md]: Revised: `The metadata fields `_index`, `_version`, `_version_type`, and `_id` cannot be removed. If `version` is specified, `_id` cannot be removed from the ingested document.`

**Pull request merged**